### PR TITLE
[saiplayer] Convert saiplayer to static library

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

**Comment by kcudnik** [comment]
retest this please

File: saiplayer/Makefile.am [hunk 1]
```diff
@@ -8,6 +8,15 @@ else
...
+						 SaiPlayer.cpp
+
+
+libSaiPlayer_a_CPPFLAGS = $(DBGFLAGS) $(AM_CPPFLAGS) $(CFLAGS_COMMON) -std=c++14
+
 saiplayer_SOURCES = saiplayer.cpp
-saiplayer_CPPFLAGS = $(DBGFLAGS) $(AM_CPPFLAGS) $(CFLAGS_COMMON)
-saiplayer_LDADD = -lhiredis -lswsscommon -lpthread -L$(top_srcdir)/meta/.libs -lsaimetadata -lsaimeta -L$(top_srcdir)/lib/src/.libs -lsairedis
+saiplayer_CPPFLAGS = $(DBGFLAGS) $(AM_CPPFLAGS) $(CFLAGS_COMMON) -std=c++14
+saiplayer_LDADD = libSaiPlayer.a ../syncd/libSyncd.a ../lib/src/libSaiRedis.a -lhiredis -lswsscommon -lpthread -L$(top_srcdir)/meta/.libs -lsaimetadata -lsaimeta
```
Comment (lguohan) [inline saiplayer/Makefile.am:22]: ../syncd/libSyncd.a looks like this breaks the lgtm build process, @kcudnik, can you check?

Comment (kcudnik) [reply saiplayer/Makefile.am:22]: it needs #599 which is already merged, how to restart lgtm to check again ?

**Comment by qiluo-msft** [comment]
retest this please

File: saiplayer/Makefile.am (hunk 1 above)
Comment (lguohan) [reply saiplayer/Makefile.am:22]: can you do a rebase and push?

Comment (lguohan) [reply saiplayer/Makefile.am:22]: image

Comment (kcudnik) [reply saiplayer/Makefile.am:22]: re run somehow don;t work when i click it

**Pull request merged**